pkg/ovs/openflow/ofctrl_builder.go

```diff
@@ -38,6 +38,17 @@ func (b *ofFlowBuilder) MatchTunMetadata(index int, data uint32) FlowBuilder {
 	return b
 }
 
+// MatchVLAN matches the VLAN VID. It holds the VLAN VID in its least significant 12 bits.
+func (b *ofFlowBuilder) MatchVLAN(nonVLAN bool, vlanId uint16, vlanMask *uint16) FlowBuilder {
```

Comment:
vlanId -> vlanID

Reply:
What does NonVLAN mean? How it is different from VLAN 0.

Do you mean we can have a 802.1Q tag with VID 0? Do we have a case to match NonVLAN in Antrea?

Reply:
Fixed.
`nonVLAN` means to match a packet without 802.1q header (untagged packet). It is not the same as VID 0, thus I use a separate parameter for this, although we don't support tagged VID=0 currently.

Reply:
`MatchVLAN` is used in `pipeline.go`. nonVLAN is to match traffic for NodeIPAM Pods and AntreaIPAM Pods without VLAN.

Reply:
Is it a valid packet to have a 802.1q header with VID 0? I feel better to define the func as using VLAN 0 to indicate nonVLAN, or add a new func (e.g. MatchNonVLAN) to match nonVLAN.

Reply:
Yes, VID=0 is valid, but this case is not supported by Antrea.
`The values of 0 and 4095 are reserved. All other values may be used as VLAN identifiers, allowing up to 4,094 VLANs. The reserved value 0x000 indicates that the frame does not carry a VLAN ID; in this case, the 802.1Q tag specifies only a priority (in PCP and DEI fields) and is referred to as a priority tag. The VID value 0xFFF is reserved for implementation use; it must not be configured or transmitted.`

Reply:
To use a combined function for this can make `flowBuilder` expression simpler.
If we don't want to support VID=0 in future, we can use vlanID=0 to indicate nonVLAN.

Reply:
Do not get why adding MatchNonVLAN will make flowBuilder expression complex. I meant add a new func of flowBuilder called MatchNonVLAN(). Do not mean we must change the match definition.

But use VLAN 0 to indicate nonVLAN also works for me.

Reply:
Replied at https://github.com/antrea-io/antrea/pull/3247/files#r819203658